pkgs/desktops/cinnamon/nemo/default.nix

```diff
@@ -0,0 +1,63 @@
+{ fetchFromGitHub
+, glib
+, gobject-introspection
+, meson
+, ninja
+, pkgconfig
+, stdenv
+, wrapGAppsHook
+, libxml2
+, cmake
+, gtk3
+, libnotify
+, cinnamon-desktop
+, xapps
+, libexif
+, exempi
+, intltool
+}:
+
+stdenv.mkDerivation rec {
+  pname = "nemo";
+  version = "4.4.1";
+  enableParallelBuilding = true;
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = pname;
+    rev = version;
+    sha256 = "0sskq0rssxvna937md446x1489hkhxys1zq03hvl8asjqa259w2q";
+  };
+
+  patches = [ ./0001-patch-datadir.patch ./0002-patch-remove-update-mime-db-script.patch ];
+
+  NIX_CFLAGS_COMPILE = [ "-I${glib.dev}/include/gio-unix-2.0" ]; # TODO: https://github.com/NixOS/nixpkgs/issues/36468
```

Comment:
Could this be passed as a meson argument, instead of via the low-level `NIX_CFLAGS_COMPILE` hack?

Reply:
I suppose. I've just copied it from the other packages.

How does one do that? Just mesonOptions = [ "-I... ?

Reply:
@nh2 It's so everywhere now that we've just gone with that.

Reply:
Also `NIX_CFLAGS_COMPILE` should be a string, not a list.

Reply:
> How does one do that? Just mesonFlags = [ "-I... ?

@mkg20001 The following works, I have just tried it:

```diff
--- a/pkgs/desktops/cinnamon/nemo/default.nix
+++ b/pkgs/desktops/cinnamon/nemo/default.nix
@@ -38,8 +38,6 @@ stdenv.mkDerivation rec {
     ./0002-patch-remove-update-mime-db-script.patch
   ];
 
-  NIX_CFLAGS_COMPILE = [ "-I${glib.dev}/include/gio-unix-2.0" ]; # TODO: https://github.com/NixOS/nixpkgs/issues/36468
-
   buildInputs = [
     glib
     gtk3
@@ -61,6 +59,11 @@ stdenv.mkDerivation rec {
     intltool
   ];
 
+  mesonFlags = [
+    # TODO: https://github.com/NixOS/nixpkgs/issues/36468
+    "-Dc_args=-I${glib.dev}/include/gio-unix-2.0"
+  ];
+
   meta = with stdenv.lib; {
     homepage = "https://github.com/linuxmint/nemo";
     description = "File browser for Cinnamon";
```

> It's so everywhere now that we've just gone with that.

@worldofpeace Understood.

I think `-Dc_args` is better and we should switch to it because

* that way all parts of meson can see the flag, passing it to potentially more places than we do with our low-level hack
* the flag shows up in the build logs, making it clear what's happening

Reply:
Thanks, I think i can mark this resolved :+1:

Reply:
I've filed #79303 for the general `NIX_CFLAGS_COMPILE` issue.